src/random-seed/random-seed.c

```diff
@@ -87,20 +87,17 @@ static CreditEntropy may_credit(int seed_fd) {
         /* Don't credit the random seed if we are in first-boot mode, because we are supposed to start from
          * scratch. This is a safety precaution for cases where we people ship "golden" images with empty
          * /etc but populated /var that contains a random seed. */
-        if (access("/run/systemd/first-boot", F_OK) < 0) {
-
-                if (errno != ENOENT) {
-                        log_warning_errno(errno, "Failed to check whether we are in first-boot mode, not crediting entropy: %m");
-                        return CREDIT_ENTROPY_NO_WAY;
-                }
-
-                /* If ENOENT all is good, we are not in first-boot mode. */
-        } else {
+        r = RET_NERRNO(access("/run/systemd/first-boot", F_OK));
+        if (r >= 0) {
                 log_debug("Not crediting entropy, since booted in first-boot mode.");
                 return CREDIT_ENTROPY_NO_WAY;
         }
+        if (r == -ENOENT)
+                /* All is good, we are not in first-boot mode. */
+                return CREDIT_ENTROPY_YES_PLEASE;
 
-        return CREDIT_ENTROPY_YES_PLEASE;
+        log_warning_errno(r, "Failed to check whether we are in first-boot mode, not crediting entropy: %m");
+        return CREDIT_ENTROPY_NO_WAY;
 }
```

Comment:
we usually want that the "clean" code paths is not the indented one, and the exceptional code path the indented one. Hence:

```c
if (r == -ENOENT)
        …
if (r < 0)
        …

…
```

Reply:
Agreed but in this specific case isn't the "clean" code path (or the common one) the path that returns `CREDIT_ENTROPY_YES_PLEASE` ?

The other code paths are either due to an error of access() or during first boot (which happens only once).